handler/order/order.go

```diff
@@ -26,18 +46,50 @@ func CreateOrder(orderz core.OrderService) http.HandlerFunc {
 			return
 		}
 
-		if body.Amount.LessThanOrEqual(decimal.Zero) {
-			render.Error(w, http.StatusBadRequest, errors.New("invalid amount"))
-			return
-		}
+		ret := make(map[string]any)
+		if body.Channel == "lemon" {
+			var chosenVariant config.TopupVariant
+			for _, v := range variants {
+				if body.VariantID == v.LemonID {
+					chosenVariant = v
+					break
+				}
+			}
+
+			if chosenVariant.LemonID == 0 {
+				render.Error(w, http.StatusBadRequest, nil)
+				return
+			}
+
+			paymentURL, err := orderz.CreateLemonOrder(ctx, user.ID, lemon.StoreID, chosenVariant.LemonID, decimal.NewFromFloat(chosenVariant.Amount), body.RedirectURL)
+			if err != nil {
+				render.Error(w, http.StatusInternalServerError, err)
+				return
+			}
+			ret["payment_url"] = paymentURL
+
+		} else if body.Channel == "mixpay" {
+			if body.Amount.LessThanOrEqual(decimal.Zero) {
+				render.Error(w, http.StatusBadRequest, errors.New("invalid amount"))
+				return
+			}
+
+			code, err := orderz.CreateMixpayOrder(ctx, user.ID, body.Amount)
+			if err != nil {
+				render.Error(w, http.StatusInternalServerError, err)
+				return
+			}
+			ret["code"] = code
```

Comment:
better also `ret["payment_url"]` here and I'll fix this.

Reply:
diff: https://github.com/pandodao/botastic/pull/70/commits/99b8d3c0946716c72d6714f08b7b87b11c08b2e8